via/static/scripts/video_player/components/VideoPlayerApp.tsx

```diff
@@ -61,19 +105,18 @@ export default function VideoPlayerApp({
           </Button>
           <div className="flex-grow" />
           <Input
-            aria-label="Filter transcript"
+            aria-label="Transcript filter"
             data-testid="filter-input"
-            onInput={e =>
-              setFilter((e.target as HTMLInputElement).value.trim())
-            }
+            onInput={e => setFilter((e.target as HTMLInputElement).value)}
             placeholder="Search..."
+            value={filter}
```

Comment:
This ensures that clearing the filter state internally also updates the input control.